pkg/generate_test.go

```diff
@@ -676,17 +676,238 @@ func TestGenerateRoute(t *testing.T) {
 	}
 }
 
-func TestGenerateOverlays(t *testing.T) {
-	component := gitopsv1alpha1.GeneratorOptions{
-		Name: "test-component",
+func TestGenerateIngress(t *testing.T) {
+	applicationName := "test-application"
+	componentName := "test-component"
+	namespace := "test-namespace"
+	customK8sLabels := map[string]string{
+		"app.kubernetes.io/name":       componentName,
+		"app.kubernetes.io/instance":   "ComponentCRName",
+		"app.kubernetes.io/part-of":    applicationName,
+		"app.kubernetes.io/managed-by": "kustomize",
+		"app.kubernetes.io/created-by": "GitOps Generator Test",
+	}
+	k8slabels := map[string]string{
+		"app.kubernetes.io/name":       componentName,
+		"app.kubernetes.io/instance":   componentName,
+		"app.kubernetes.io/part-of":    applicationName,
+		"app.kubernetes.io/managed-by": "kustomize",
+		"app.kubernetes.io/created-by": "application-service",
+	}
+	implementationSpecific := networkingv1.PathTypeImplementationSpecific
+
+	tests := []struct {
+		name        string
+		options     gitopsv1alpha1.GeneratorOptions
+		wantIngress networkingv1.Ingress
+	}{
+		{
+			name: "Simple options object",
+			options: gitopsv1alpha1.GeneratorOptions{
+				Name:        componentName,
+				Namespace:   namespace,
+				Application: applicationName,
+				TargetPort:  5000,
```

Comment:
Can TargetPort be 0?  Do we need test coverage for that?

Reply:
good question, it can be 0 from this library's perspective but probably not from RHTAP since we will always have a default. 

There is a check for this in https://github.com/redhat-developer/gitops-generator/pull/45/files#diff-afe63b2c70508656d0bfa0f0acb5d9bfc188b3e65ab7bf007f6e0b2cf52ad46fR199